command/var_list.go

```diff
@@ -0,0 +1,276 @@
+		return 1
+	}
+
+	if len(args) == 1 {
+		prefix = args[0]
+	}
+
+	// Get the HTTP client
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	if filter != "" {
+		c.Ui.Warn(msgWarnFilterPerformance)
+	}
+
+	qo := &api.QueryOptions{
+		Filter:    filter,
+		PerPage:   int32(perPage),
+		NextToken: pageToken,
+		Params:    map[string]string{},
+	}
+
+	vars, qm, err := client.SecureVariables().PrefixList(prefix, qo)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error retrieving vars: %s", err))
+		return 1
+	}
+
+	switch {
+	case json:
+
+		// obj and items enable us to rework the output before sending it
+		// to the Format method for transformation into JSON.
+		var obj, items interface{}
+		obj = vars
+		items = vars
+
+		if quiet {
+			items = itemsToQuietJSONReadySlice(vars, c.Meta.namespace)
+			obj = items
+		}
+
+		// If the response is paginated, we need to provide a means for the
+		// caller to get to the pagination information. Wrapping the list
+		// in a struct for the special case allows this extra data without
+		// adding unnecessary structure in the non-paginated case.
+		if perPage > 0 {
+
+			obj = struct {
+				Data      interface{}
+				QueryMeta *api.QueryMeta
+			}{
+				items,
+				qm,
+			}
+		}
+
+		// By this point, the output is ready to be transformed to JSON via
+		// the Format func.
+		out, err := Format(json, tmpl, obj)
+		if err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+
+		c.Ui.Output(out)
+
+		// Since the JSON formatting deals with the pagination information
+		// itself, exit the command here so that it doesn't double print.
+		return 0
+
+	case quiet:
+		c.Ui.Output(
+			formatList(
+				itemsToQuietStringSlice(vars, c.Meta.namespace)))
+
+	case len(tmpl) > 0:
+		out, err := Format(json, tmpl, vars)
+		if err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+
+		c.Ui.Output(out)
+
+	default:
+		c.Ui.Output(formatVarStubs(vars))
+	}
+
+	if qm.NextToken != "" {
+		// This uses Ui.Warn to output the next page token to stderr
+		// so that scripts consuming paths from stdout will not have
+		// to special case the output.
+		c.Ui.Warn(fmt.Sprintf("Next page token: %s", qm.NextToken))
+	}
+
+	return 0
+}
+
+func formatVarStubs(vars []*api.SecureVariableMetadata) string {
+	if len(vars) == 0 {
+		return msgSecureVariableNotFound
+	}
+
+	// Sort the output by variable namespace, path
+	sort.Slice(vars, func(i, j int) bool {
+		if vars[i].Namespace == vars[j].Namespace {
+			return vars[i].Path < vars[j].Path
+		}
+		return vars[i].Namespace < vars[j].Namespace
+	})
+
+	rows := make([]string, len(vars)+1)
+	rows[0] = "Namespace|Path|Last Updated"
+	for i, sv := range vars {
+		rows[i+1] = fmt.Sprintf("%s|%s|%s",
+			sv.Namespace,
+			sv.Path,
+			time.Unix(0, sv.ModifyTime),
+		)
+	}
+	return formatList(rows)
+}
+
+func itemsToQuietStringSlice(vars []*api.SecureVariableMetadata, ns string) []string {
```

Comment:
Nitpick: calling these functions "itemsTo..." overloads the usage of the field name `Items`, and this would be a little easier to read if we avoided it.